transmogrifier/sources/transformer.py

```diff
@@ -153,233 +208,308 @@ def parse_source_file(cls, source_file: str) -> Iterator[JSON | Tag]:
 
     @classmethod
-    @abstractmethod
-    def get_source_record_id(cls, source_record: JSON | Tag) -> str:
+    def get_main_titles(cls, source_record: JSON | Tag) -> list[str]:
         """
-        Get or generate a source record ID from a source record.
+        Retrieve main title(s) from an source record.
 
         Must be overridden by source subclasses.
 
         Args:
             source_record: A single source record.
         """
-        return ""
+        return []
 
     @classmethod
     @abstractmethod
-    def record_is_deleted(cls, source_record: JSON | Tag) -> bool:
+    def get_source_link(
+        cls, source_base_url: str, source_record_id: str, source_record: JSON | Tag
+    ) -> str:
         """
-        Determine whether record has a status of deleted.
+        Class method to set the source link for the item.
 
         Must be overridden by source subclasses.
 
         Args:
+            source_base_url: Source base URL.
+            source_record_id: Record identifier for the source record.
             source_record: A single source record.
         """
-        return False
+        pass
 
+    @classmethod
     @abstractmethod
-    def get_required_fields(self, source_record: JSON | Tag) -> dict:
+    def get_timdex_record_id(
+        cls, source: str, source_record_id: str, source_record: Tag
+    ) -> str:
         """
-        Get required TIMDEX fields from a source record.
+        Class method to set the TIMDEX record id.
 
-        Must be overridden by format subclasses.
+        Must be overridden by source subclasses.
 
         Args:
+            source: Source name.
+            source_record_id: Record identifier for the source record.
             source_record: A single source record.
         """
-        return {}
+        pass
 
+    @classmethod
     @abstractmethod
-    def transform(self, source_record: JSON | Tag) -> Optional[TimdexRecord]:
+    def get_source_record_id(cls, source_record: JSON | Tag) -> str:
         """
-        Transform a source record into a TIMDEX record.
+        Get or generate a source record ID from a source record.
 
-        Must be overridden by format subclasses.
+        Must be overridden by source subclasses.
 
         Args:
             source_record: A single source record.
         """
-        return None
+        pass
 
     @classmethod
     @abstractmethod
-    def get_valid_title(cls, source_record_id: str, source_record: JSON | Tag) -> str:
+    def record_is_deleted(cls, source_record: JSON | Tag) -> bool:
         """
-        Retrieves main title(s) from a source record and returns a valid title string.
+        Determine whether record has a status of deleted.
 
         Must be overridden by source subclasses.
 
         Args:
-            source_record_id: Record identifier for the source record.
             source_record: A single source record.
         """
-        return ""
+        pass
 
-    @classmethod
-    @abstractmethod
```

Comment:
This feels correct to me, having `get_optional_fields()` **not** be required by any children of this base `Transformer` class, meaning they will receieve an empty dictionary of optional fields if they don't define this method.